### DISCUSSION TITLE
Reporte

### DISCUSSION THREAD

**Discussion by gklimm** [te-online/timemanager#164] [Ideas]
Hallo lieber Thomas,

da viele auch die Stundenaufstellungen als Anhang zu Rechnungen mit versenden müssen, wäre meine Idee den CSV-Export in der NC gleich an Lool-Calc zu übergeben mit einer entsprechenden Vorlage, die man einstellt oder beim Aufruf auswählen kann.

**Comment by te-online** [comment]
Hallo! Ich bin mir nicht sicher, ob ich dich verstehe 😅 

Mit Lool meinst du Libre Office? Man könnte eventuell einen Knopf machen, der die CSV-Datei gleich in Nextcloud ablegt und (sofern man Collabora o.ä. installiert hat) kann man die Datei dann gleich öffnen.

Weiß nicht, was du mit Vorlage meinst?

**Comment by gklimm** [comment]
Ja, korrekt Libreoffice Online ist gemeint. Bei Erstellung einer neuen Datei kann man eine Vorlage aus dem Vorlagenverzeichnis auswählen. Hier wird sogar ein Stundenzettel für das ganze Jahr als Vorlage mitgeliefert mit der App oder NC.
Mein Ansatz war einfach die "vorhandenen" Dinge zu nutzen, bevor man einen Reportgenerator (am Ende artet das immer aus) baut.

**Comment by te-online** [comment]
Ich habe LibreOffice Online noch nie benutzt, das würde also leider so oder so recht aufwändig sein, mich da einzuarbeiten. Aber wenn du mehr Infos hast, bin ich offen für Integrationen 😊 Was müsste man im Code ändern? Gibt es eine Schnittstelle für die LibreOffice app?

**Comment by gklimm** [comment]
Es ist leider schon Jahre her als ich mal via uno (die Schnittstelle von LibreOffice, damals glaube ich noch StarOffice) Daten in die Applikation gepumpt habe.
Ansatz war eine möglichst einfache Variante um eine Auswertung zu erhalten, die man dem Kunden als Anlage zur Rechnung mitschicken kann. Nicht jeder Kunde wird über eine csv begeistert sein 😉
Bei einem kleinen "Reportgenerator" sehe ich halt immer das Faß ohne Boden. Wenn der Benutzer das über die Tabellenkalkulation selbst machen kann, hast Du das von der Backe.

**Reply by te-online** [reply]
Verstehe 🙂  Ich bin mir leider nicht sicher, wie ich helfen kann? Ich kenne die uno Schnittstelle nicht und hab auch nicht so richtig Bedarf für dieses Feature. Du kann in der Datenbank die Tabellen ja mal inspizieren oder dir das CSV genauer ansehen. Vielleicht kann man per Browser-Erweiterung, Automatisierungsservice o.ä. einen Konverter bauen, der die CSV-Datei als Input nimmt?

**Reply by gklimm** [reply]
https://www.openproject.org/docs/user-guide/time-and-costs/reporting/ liefert eine Tabelle, die ich dann auch an die Rechnung für den Kunden hängen kann ...

**Reply by te-online** [reply]
Wenn ich die Tabelle aus dem TimeManager CSV Export mit dem Screenshot vom Openproject Export vergleiche, sind diese sich doch sehr ähnlich. Was würde deiner Ansicht nach fehlen?

<img width="1796" alt="Screenshot 2024-12-21 at 19 34 42" src="https://github.com/user-attachments/assets/6a035451-5798-4d0d-bbf9-1c999488753a" />
